src/main/java/org/spongepowered/api/Game.java

```diff
@@ -217,7 +260,9 @@ default ServiceManager getServiceManager() {
      * Gets the scheduler used to schedule tasks.
      *
      * @return The scheduler
+     * @deprecated Use {@link Server#getScheduler()} or {@link Client#getScheduler()}
```

Comment:
Why not make it based on current thread?

Reply:
Because it shouldn't be a guessing game - someone should explicitly request which scheduler they want.

Reply:
Right, it shouldn't be a guessing game, but it should also be able to detect which active thread is currently used based on the caller. This should be as easy as possible to use personally, and yes exposing the client and server, a majority of plugin developers will not know to use the client, until it's seemingly necessary in a corner case their plugin is installed on the client (like say VoxelSniper). All of the code should still work on the client regardless whether it was written in mind to be server side, because the simple call to `Sponge.getScheduler()` at the *call site* is on the server thread already.

Reply:
I think I like gabizou's idea to get a Scheduler (An Optional one) that fits the current thread you are on.

I don't think this is the appropriate place for it and the Scheduler here does need to be removed.

In the Sponge god object would be okay imo.

Reply:
Actually I want to take some time to weigh the arguments on both sides and ponder this.

Reply:
It's fine for schedulers on the server/client thread, but what about if I wanted to get a sync scheduler from an async thread to perform a block change after a database query? Unless you track which scheduler created which thread, if you remove the ability to select the correct scheduler, then it becomes a guessing game once again.

I think we need to be careful about making assumptions about things without providing a credible way to override them. Assumptions are not going to work 100% of the time, and if someone find some way to use SF for a client/internal-server hybrid (if that's even possible, I don't know!), then there might be a use case?

We should always be giving developers the chance to be explicit about what they want when it makes sense to. I would say that this is one of those times - if a plugin developer is developing for the client, they know that's the case! However, we also need through docs about where, for an integrated server, the server scheduler and the client scheduler should be used, because providing both is possibly going to muddy the waters for developers, I fear. I get that making assumptions would remove that need somewhat, but removing the chance to be explicit will bite us at some point.

With that in mind, what about `Optional<Scheduler> Game#getScheduler(SchedulerType);` where `SchedulerType` is a catalog type and is either `SERVER` or `CLIENT`?

Reply:
I’d wager the Scheduler can have a platform since we expose the platform type. So you can determine if there’s a client or server scheduled after querying for it from the god Object. This way you can determine which type you have to properly schedule a synchronized task.

Reply:
I've thought about it and we're sticking with the deprecation.

Its not hard to grab either Client or Server objects and call getScheduler....

Reply:
Hmm, what if a plugin wants the scheduler in loading stages? In that case, the scheduler would point to one that exists (on client the one points to client one, on dedicated server the one points to the server one)